optuna_dashboard/ts/components/DataGrid.tsx

```diff
@@ -239,30 +190,119 @@ function DataGrid<T>(props: {
   )
 }
 
+const TableHeaderCellSpan = styled("span")({
+  display: "inline-flex",
+})
+
+const HiddenSpan = styled("span")({
+  border: 0,
+  clip: "rect(0 0 0 0)",
+  height: 1,
+  margin: -1,
+  overflow: "hidden",
+  padding: 0,
+  position: "absolute",
+  top: 20,
+  width: 1,
+})
+
+function DataGridHeaderColumn<T>(props: {
+  column: DataGridColumn<T>
+  order: Order | null
+  onOrderByChange: (order: Order) => void
+  filter: RowFilter | null
+  onFilterChange: (values: Value[]) => void
+  dense?: boolean
+}) {
+  const { column, order, onOrderByChange, filter, onFilterChange, dense } =
+    props
+  const [filterMenuAnchorEl, setFilterMenuAnchorEl] =
+    React.useState<null | HTMLElement>(null)
+
+  const filterChoices = column.filterChoices
+
+  return (
+    <TableCell
+      padding={column.padding || "normal"}
+      sortDirection={order !== null ? order : false}
+    >
+      <TableHeaderCellSpan>
+        {column.sortable ? (
+          <TableSortLabel
+            active={order !== null}
+            direction={order || "asc"}
+            onClick={() => {
+              onOrderByChange(order === "asc" ? "desc" : "asc")
+            }}
+          >
+            {column.label}
+            {order !== null ? (
+              <HiddenSpan>
+                {order === "desc" ? "sorted descending" : "sorted ascending"}
+              </HiddenSpan>
+            ) : null}
+          </TableSortLabel>
+        ) : (
+          column.label
+        )}
+        {filterChoices !== undefined ? (
+          <>
+            <IconButton
+              size={dense ? "small" : "medium"}
+              onClick={(e) => {
+                setFilterMenuAnchorEl(e.currentTarget)
+              }}
+            >
+              <FilterListIcon fontSize="small" />
+            </IconButton>
+            <Menu
+              anchorEl={filterMenuAnchorEl}
+              open={filterMenuAnchorEl !== null}
+              onClose={() => {
+                setFilterMenuAnchorEl(null)
+              }}
+            >
+              {filterChoices.map((choice) => (
+                <MenuItem
+                  key={choice}
+                  onClick={() => {
+                    const values =
+                      filter === null
+                        ? filterChoices.filter((v) => v !== choice)
+                        : filter.values.some((v) => v === choice)
+                        ? filter.values.filter((v) => v !== choice)
+                        : [...filter.values, choice]
+                    onFilterChange(values)
```

Comment:
To apply https://github.com/optuna/optuna-dashboard/pull/700/files#r1398699656.

```suggestion
                    const newTickedValues =
                      filter === null
                        ? filterChoices.filter((v) => v !== choice)
                        : filter.values.some((v) => v === choice)
                        ? filter.values.filter((v) => v !== choice)
                        : [...filter.values, choice]
                    onFilterChange(newTickedValues)
```